src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -75,8 +77,21 @@ using namespace ::chip;
 using namespace ::chip::Ble;
 using namespace ::chip::DeviceLayer::Internal;
 
+void sl_get_bluethooth_addr(uint8_t * addr, uint8_t length)
+{
+    uint16_t randNumber = chip::Crypto::GetRandU16();
+    srand(randNumber);
+    while (length > 0)
+    {
+        int value        = std::rand();
```

Comment:
chip::Crypto::GetRandU16(); already provide a random number. I am not sure why you want to use that random number as seed and then do std:rand() again.

Wouldn't it work directly with `GetRandU8();` for each byte of your array?  a single for loop with a one-liner would work.
```
for (i=0 i < nbBytes; i++)
{
    addr[i] = GetRandU8();
}
```

Reply:
removed rand function using the crypto GetRandU8 function